### PR TITLE
PR: Limit the number of flags in the editor

### PR DESCRIPTION
**Pull request by impact27**
## Description of Changes
If there are too many flags in the document, the flags will slow down the printing. This disables the flags of a type if there are more than 10000.

To reproduce, create a file with:
```
with open("test.py", "w") as f:
    for i in range(10000):
        f.write("aaaaaa\n")
```
This will create 10000 error flags, which will slow down spyder significantly

* [ ] Wrote at least one-line docstrings (for any new functions)
* [ ] Added unit test(s) covering the changes (if testable)

* [ ] Included a screenshot or animation (if affecting the UI, see [Licecap](https://www.cockos.com/licecap/))

### Issue(s) Resolved

Fixes #

### Affirmation

By submitting this Pull Request or typing my (user)name below,
I affirm the [Developer Certificate of Origin](https://developercertificate.org)
with respect to all commits and content included in this PR,
and understand I am releasing the same under Spyder's MIT (Expat) license.

I certify the above statement is true and correct:

### CODE REVIEW

File: spyder/plugins/editor/panels/scrollflag.py [hunk 1]
```diff
@@ -23,6 +23,8 @@
 
 
 REFRESH_RATE = 1000
+# Maximum number of flags to print in a file
```
Comment (ccordoba12) [inline spyder/plugins/editor/panels/scrollflag.py:26]:
```suggestion

# Maximum number of flags to print in a file
```

File: spyder/plugins/editor/panels/scrollflag.py [hunk 2]
```diff
@@ -216,14 +213,41 @@ def compute_flag_ypos(block):
...
-                painter.drawRect(rect_x, rect_y, rect_w, rect_h)
+            if editor.verticalScrollBar().maximum() == 0:
+                # No scroll
+                for block in dict_flag_lists[flag_type]:
+                    if not block.isValid():
+                        continue
+                    geometry = editor.blockBoundingGeometry(block)
+                    rect_y = ceil(
+                        geometry.y()
+                        + geometry.height() / 2 + rect_h / 2)
```
Comment (ccordoba12) [inline spyder/plugins/editor/panels/scrollflag.py:224]:
```suggestion
                    rect_y = ceil(
                        geometry.y() +
                        geometry.height() / 2 +
                        rect_h / 2
                    )
```

File: spyder/plugins/editor/panels/scrollflag.py [hunk 3]
```diff
@@ -216,14 +213,41 @@ def compute_flag_ypos(block):
...
+                    painter.drawRect(rect_x, rect_y, rect_w, rect_h)
+
```
Comment (ccordoba12) [inline spyder/plugins/editor/panels/scrollflag.py:226]:
```suggestion
```

**Review by ccordoba12 — COMMENTED** [review]
Thanks @impact27 for your help with this! I left some minor style suggestions, otherwise looks good to me.

**Comment by ccordoba12** [comment]
@impact27, one thing I forgot to mention: could you add a test for this? Just to check that we're not trying to display more than `MAX_FLAGS` in the scrollflag area.

**Comment by impact27** [comment]
I am not sure how to do that. the paint code does a bunch of "painter.drawRect" but I think the result is just an image. I wouldn't know how to check how many rectangles are drawn.

**Comment by ccordoba12** [comment]
What if you check the length of `self._dict_flag_list[flag_type]` for a certain flag type that goes over `MAX_FLAGS` in a file?

**Comment by impact27** [comment]
The filtering is applied at the print stage, so the dictionary would contain more than the limit of elements

**Comment by ccordoba12** [comment]
Ok, no problem then. Creating a test for this is way harder than I thought.

**Review by ccordoba12 — APPROVED** [review]
Thanks @impact27!

**Pull request merged**